lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -44,43 +57,82 @@ local function playerIsInBuild( ply )
     return !getPlayerPvpMode( ply )
 end
 
-local function movementMultiplier(weaponCount) 
-    return math.Clamp(1 - (1.9^( weaponCount))/100, 0, 1)
+local minMult = 0.2
+local function movementMultiplier( weaponCount )
+    local N = 1 - (1.9^( weaponCount ))/100
+    return math.Clamp( N, 0.5, 1 )
 end
 
 local function setSpeed(ply, multiplier) 
     -- set speed multiplier, 0 - 1
-    ply:SetRunSpeed(math.Clamp(baseRunSpeed*multiplier,minRunSpeed, baseRunSpeed))
-    ply:SetWalkSpeed(math.Clamp(baseWalkSpeed*multiplier,minWalkSpeed, baseWalkSpeed))
+    ply:SetRunSpeed( math.Clamp( baseRunSpeed*multiplier, minRunSpeed, baseRunSpeed ) )
+    ply:SetWalkSpeed( math.Clamp( baseWalkSpeed*multiplier,minWalkSpeed, baseWalkSpeed ) )
 end
 
-local function adjustMovementSpeed(ply) 
+function adjustMovementSpeed(ply) 
     if playerIsInBuild( ply ) then 
-        setSpeed(ply, 1) 
+        setSpeed( ply, 1 ) 
         return
     end
     
     local weapons = ply:GetWeapons()
     local wepCount = 0
     
     -- count weapons
-    for k, weapon in pairs(weapons) do
+    for _, weapon in pairs( weapons ) do
```

Comment:
👍

Reply:
Perhaps another option would be to compare the tables of current weapons and unaffected weapons and count the difference. Is that called intersection?

Reply:
I don't think there's a table.intersect